src/debuginfo.cpp

```diff
@@ -1277,50 +1277,61 @@ int jl_getFunctionInfo(jl_frame_t **frames_out, size_t pointer, int skipC, int n
         std::vector<JITEvent_EmittedFunctionDetails::LineStart>::iterator vit =
             (*it).second.lines.begin();
         JITEvent_EmittedFunctionDetails::LineStart prev = *vit;
-
-        if ((*it).second.func) {
-            DISubprogram debugscope =
-                DISubprogram(prev.Loc.getScope((*it).second.func->getContext()));
-            jl_copy_str(&frames[0].file_name, debugscope.getFilename().str().c_str());
-            // the DISubprogram has the un-mangled name, so use that if
-            // available. However, if the scope need not be the current
-            // subprogram.
-            if (debugscope.getName().data() != NULL) {
-                jl_copy_str(&frames[0].func_name, debugscope.getName().str().c_str());
-            }
-            else {
-                char *oldname = frames[0].func_name;
-                frames[0].func_name = jl_demangle(frames[0].func_name);
-                free(oldname);
-            }
+        LLVMContext &Ctx = (*it).second.func->getContext();
+
+        DISubprogram debugscope(prev.Loc.getScope(Ctx));
+        jl_copy_str(&frames[0].file_name, debugscope.getFilename().str().c_str());
+        // the DISubprogram has the un-mangled name, so use that if
+        // available. However, if the scope need not be the current
```

Comment:
grammar makes no sense here - what is "if the scope need not be the current subprogram" trying to say?